### PR TITLE
Fix: TSH Updates & Emitted Txn

### PR DESCRIPTION
**Pull request by dangell7**
### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (you added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation update
- [ ] Chore (no impact to binary, e.g. `.gitignore`, formatting, dropping support for older tooling)
- [ ] Release

### API Impact

- [ ] Public API: New feature (new methods and/or new fields)
- [ ] Public API: Breaking change (in general, breaking changes should only impact the next api_version)
- [ ] `libxrpl` change (any change that may affect `libxrpl` or dependents of `libxrpl`)
- [ ] Peer protocol change (must be backward compatible or bump the peer protocol version)

### CODE REVIEW

**Review by tequdev — COMMENTED** [review]
Some sections (ttTRUST_SET,ttNFTOKEN_CREATE/ACCEPT_OFFER,ttPAYCHAN_CLAIM, etc.) seem to be replacing canRollback with tshWEAK instead of tshSTRONG without fixXahauV2 checking. Is this OK?

**Comment by RichardAH** [comment]
> Some sections (ttTRUST_SET,ttNFTOKEN_CREATE/ACCEPT_OFFER,ttPAYCHAN_CLAIM, etc.) seem to be replacing canRollback with tshWEAK instead of tshSTRONG without fixXahauV2 checking. Is this OK?

canRollback was derived from a static table in applyHook.h, so all these replacements should be identical to what they already were in that table, but please feel free to check

File: src/ripple/app/hook/impl/applyHook.cpp [hunk 1]
```diff
@@ -228,16 +254,16 @@ getTransactionalStakeHolders(STTx const& tx, ReadView const& rv)
 
             if (bo)
             {
-                ADD_TSH(bo->getAccountID(sfOwner), canRollback);
+                ADD_TSH(bo->getAccountID(sfOwner), tfStrongTSH);
```
Comment (RichardAH) [inline src/ripple/app/hook/impl/applyHook.cpp]: fix this

File: src/ripple/app/hook/impl/applyHook.cpp [hunk 2]
```diff
@@ -228,16 +254,16 @@ getTransactionalStakeHolders(STTx const& tx, ReadView const& rv)
...
                 if (bo->isFieldPresent(sfDestination))
-                    ADD_TSH(bo->getAccountID(sfDestination), canRollback);
+                    ADD_TSH(bo->getAccountID(sfDestination), tshWEAK);
             }
 
             if (so)
             {
-                ADD_TSH(so->getAccountID(sfOwner), canRollback);
+                ADD_TSH(so->getAccountID(sfOwner), tfStrongTSH);
```
Comment (RichardAH) [inline src/ripple/app/hook/impl/applyHook.cpp]: here too

File: src/ripple/app/hook/impl/applyHook.cpp [hunk 3]
```diff
@@ -253,24 +279,21 @@ getTransactionalStakeHolders(STTx const& tx, ReadView const& rv)
                 auto const offer = getNFTOffer(offerID, rv);
                 if (offer)
                 {
-                    ADD_TSH(offer->getAccountID(sfOwner), canRollback);
+                    ADD_TSH(offer->getAccountID(sfOwner), tfStrongTSH);
```
Comment (RichardAH) [inline src/ripple/app/hook/impl/applyHook.cpp]: and here

**Pull request merged**